modules/ppcp-settings/resources/js/data/onboarding/actions.js

```diff
@@ -100,3 +100,43 @@ export function refresh() {
 		select.persistentData();
 	};
 }
+
+/**
+ * Persistent. Updates the gateway synced status.
+ *
+ * @param {boolean} synced The sync status to set
+ * @return {Action} The action.
+ */
+export const updateGatewaysSynced = ( synced = true ) =>
+	setPersistent( 'gatewaysSynced', synced );
+
+/**
+ * Persistent. Updates the gateway refreshed status.
+ *
+ * @param {boolean} refreshed The refreshed status to set
+ * @return {Action} The action.
+ */
+export const updateGatewaysRefreshed = ( refreshed = true ) =>
+	setPersistent( 'gatewaysRefreshed', refreshed );
+
+/**
+ * Action creator to sync payment gateways.
+ * This will both update the state and persist it.
+ *
+ * @return {Function} The thunk function.
+ */
+export function syncGateways() {
+	return async ( { dispatch } ) => {
+		dispatch( { type: ACTION_TYPES.SYNC_GATEWAYS } );
+		await dispatch( persist() );
+		return { success: true };
+	};
```

Comment:
While this is correct, our Redux stores generally use the (badly named) helpers `setPersistent()` or `setTransient()`

A more aligned way for this action would be:

```js
// use a "const" instead of a function.
export function syncGateways() {
  return async ( { dispatch } ) => {
    setPersistent( 'gatewaysSynced', true );
    await dispatch.persist(); // note that dispatch is not a function but an object
  }
}
```